03-objects-and-composition/exercise/03. Car Factory

```diff
@@ -0,0 +1,29 @@
+function carFactory(car) {
+    let newCar = {};
```

Comment:
You can try to refactor this solution using a little bit more advanced objects composition:

```javascript
const engineInfo = {
    90: { power: 90, volume: 1800 },
    120: { power: 120, volume: 2400 },
    default: { power: 200, volume: 3500 }
  };

  const newCar = {
    model: car.model,
    engine: engineInfo[car.power <= 90 ? 90 : car.power <= 120 ? 120 : 'default'], // or extract this into separate function
    . . . TODO
```

Reply:
Thank you for your suggestion and feedback!